File: src/main/java/dev/bacongubbe/wishlistapp/user/UserService.java
package dev.bacongubbe.wishlistapp.user;

import org.springframework.stereotype.Service;

@Service
public class UserService {

    private final UserRepository repo;

    public UserService(UserRepository repo){
        this.repo = repo;
    }

    public User getUser(String id) {
        return repo.getUser(id);
    }

    public User createUser(User user){
        return repo.createUser(user);
    }
}

File: server/src/main/kotlin/dev/bacongubbe/wishlist/service/UserService.kt
package dev.bacongubbe.wishlist.service

import dev.bacongubbe.wishlist.User_entity
import dev.bacongubbe.wishlist.repo.UserRepo

class UserService(private val repo : UserRepo) {

    suspend fun getUsers(): List<User_entity> = repo.getUsers()

    suspend fun addUser(name : String, email: String) = repo.addUser(name, email)

    suspend fun deleteUser(id: String) = repo.deleteUser(id)

    suspend fun getUserById(id: String): User_entity = repo.getUserById(id)

}